### PR TITLE
BDOG-705 user revoke consent (part 2)

### PR DESCRIPTION
**Pull request by natcarey**
In this PR we have:

 - Moved all copy into the message file
 - Established the interface between the page and the Javascript
 - Bundled a JS file specifically for this page (which can rely on modules like `userPreferenceFactory`

We still need to:

 - Hook up the radio buttons as input and output

### CODE REVIEW

**Comment by cjwebb** [comment]
Looks good

File: app/uk/gov/hmrc/trackingconsentfrontend/assets/src/cookieSettingsPage.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const castToArray = nodeList => Array.prototype.slice.apply(nodeList)
+
+document.addEventListener('DOMContentLoaded', () => {
+  castToArray(document.querySelectorAll('[data-module="cookie-settings"]')).forEach(cookieSettingsForm => {
```
Comment (matthewmascord) [inline app/uk/gov/hmrc/trackingconsentfrontend/assets/src/cookieSettingsPage.ts]: You could use the array spread syntax here - it should be transpiled by Babel.

Comment (matthewmascord) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/cookieSettingsPage.ts]: i.e. [...document.querySelectorAll('[data-module="cookie-settings"]')] - I could be wrong!

Comment (natcarey) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/cookieSettingsPage.ts]: I just tried replacing `const castToArray = nodeList => Array.prototype.slice.apply(nodeList)` with `const castToArray = nodeList => [...nodeList]` and it's being simplified down to `[].concat(nodeList)` which just wraps my nodeList in an array.

Comment (matthewmascord) [reply app/uk/gov/hmrc/trackingconsentfrontend/assets/src/cookieSettingsPage.ts]: Weird, it should work.

**Pull request merged**